Python/optimizer_bytecodes.c

```diff
@@ -414,6 +414,13 @@ dummy_func(void) {
         }
     }
 
+    op(_GUARD_TOS_UNICODE, (value -- value)) {
+        if (sym_matches_type(value, &PyUnicode_Type)) {
+            REPLACE_OP(this_instr, _NOP, 0, 0);
+        }
+        sym_set_type(value, &PyUnicode_Type);
+    }
+
     op(_TO_BOOL_STR, (value -- res)) {
         if (!optimize_to_bool(this_instr, ctx, value, &res)) {
             res = sym_new_truthiness(ctx, value, true);
```

Comment:
Sorry, one more thing. On the line below this, can you remove the `sym_set_type(value, &PyUnicode_Type);` call? It's not needed anymore, since the type is already narrowed by the preceding guard op.

Reply:
Thanks, done!

Reply:
Sorry, I think you removed it from the wrong op.

`_GUARD_TOS_UNICODE` should have a `sym_set_type` call at the end, since we know it's a `str` at that point. `_TO_BOOL_STR` doesn't need a `sym_set_type` call, since, it's guarded by `_GUARD_TOS_UNICODE`.